candies/cli/parsers/standard.py

```diff
@@ -140,8 +129,51 @@ def configure_flag(x: inspect.Parameter, config: Dict):
                              f"'{short}': "
                              f"it is not an alphabetic letter")
 
-        parser.add_argument(f'--{parameter.name}',
-                            f'-{short}',
+        return short
+
+    def prefixes_of(x: Union[Type, Any]) -> Optional[str]:
+        # TODO: Add more prefixes depending on `prefix` attribute.
+        return '-', '--'
+
+    def is_(x: inspect.Parameter, of: Type) -> bool:
+        return isinstance(x, of) or x == of
+
+    def configure_arg(x: inspect.Parameter, config: Dict):
+        parser.add_argument(x.name, **config)
+
+    def configure_flag(x: inspect.Parameter, config: Dict):
+        # This actually makes an arg to behave like a flag,
+        # so one could call `--x` instead of `--x y`.
+        config['action'] = 'store_true'
+
+        if x.default is x.empty:
+            config['default'] = False
+        else:
+            config['default'] = bool(x.default)
+
+        short = short_of(x.annotation)
+        short = short if short is not None else x.name[0]
+
+        short_prefix, long_prefix = prefixes_of(x.annotation)
+
+        parser.add_argument(f'{long_prefix}{x.name}',
+                            f'{short_prefix}{short}',
+                            **config)
+
+    def configure_opt(x: inspect.Parameter, config: Dict):
+        if x.default is not x.empty:
+            config['default'] = x.default
+            config['required'] = False
+        else:
+            config['required'] = True
+
+        short = short_of(x.annotation)
+        short = short if short is not None else x.name[0]
+
+        short_prefix, long_prefix = prefixes_of(x.annotation)
+
+        parser.add_argument(f'{long_prefix}{x.name}',
+                            f'{short_prefix}{short}',
```

Comment:
I think we need to handle duplicate short names somehow. It would probably be better to totally remove implicit short names.

Reply:
Duplicated short names are handled by `argparse` itself. Consider looking at [`test_two_flags_with_same_implicit_shorts`](https://github.com/candy-kingdom/cli/blob/develop/tests/integration/flag_test.py#L265-L275) and [`test_two_flags_with_same_explicit_shorts`](https://github.com/candy-kingdom/cli/blob/develop/tests/integration/flag_test.py#L278-L288).

Do you think we need to handle them by ourselves?

Reply:
> Duplicated short names are handled by `argparse` itself.

Yes, I understand that (and I certainly remember that I asked about this previously).

However, I would not want my code to fail only because I used two arguments that start with the same letter. I think we shouldn't use short names at all unless the user explicitly specify them.

Reply:
Oh, I see. Yeah, great idea. Pushed!